src/minimum_tracer/minimum_tracer.cpp

```diff
@@ -2401,6 +2401,12 @@ Vacuum::Vacuum(const double &T_lowIn,
       PhasesList[i].id = i;
     }
 
+    // impose that true vacuum at T = Tthigh has id = 0.
+    for (auto &phase : PhasesList)
+      if ((phase.T_high == T_high) and
+          (phase.Get(T_high).potential < PhasesList[0].Get(T_high).potential))
+        std::swap(PhasesList[0], phase);
```

Comment:
Are you sure this works? You are swapping an iterator with a reference here.
Maybe use an iterator based for loop and use [iterator_swap](https://en.cppreference.com/w/cpp/algorithm/iter_swap)

Reply:
Or maybe easier:
You could use [min_element](https://en.cppreference.com/w/cpp/algorithm/min_element) with a compare function to get the position of the required element and then use iterator_swap just once.
With this there would only be one swap

Reply:
> Are you sure this works? You are swapping an iterator with a reference here. Maybe use an iterator based for loop and use [iterator_swap](https://en.cppreference.com/w/cpp/algorithm/iter_swap)

Change this so that iterators are used. Alsoa added testing for this fix.